src/json.ts

```diff
@@ -67,7 +83,27 @@ export const JsonStruct: Struct<Json> = union([
  * @returns Whether the value is a valid {@link Json} value.
  */
 export function isValidJson(value: unknown): value is Json {
-  return is(value, JsonStruct);
+  try {
+    getValidatedAndSanitizedJson(value as Json);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Validate and sanitize JSON structure.
+ *
+ * Note:
+ * This function will use structure that stringify and then parse
```

Comment:
I'm assuming it's coming in a next commit

Reply:
Not sure if I understand the comment. So, I wanted to explain the need for this function a little bit more. I updated the `JsonStruct` according to suggestions from review comments in https://github.com/MetaMask/snaps-monorepo/pull/1405 and returned `UnsafeJsonStruct` as it was. 
`JsonStruct` now has `return JSON.parse(JSON.stringify(value));`.
I change the docstring comment a little bit if that makes sense.